uber/site_sections/registration.py

```diff
@@ -97,7 +97,9 @@ def form(self, session, message='', return_to='', omit_badge='', check_in='', **
                 if check_in:
                     attendee.checked_in = localized_now()
                 session.add(attendee)
-                if return_to:
+                if params['badge_status'] == c.INVALID_STATUS:
+                    self.delete(attendee.id)
```

Comment:
Oh, you're right - and in this case we really don't want to delete the attendee when they're set to invalid status. Derp.